deepcell_tracking/utils.py

```diff
@@ -200,26 +203,88 @@ def trk_folder_to_trks(dirname, trks_filename):
 
 
 def save_trks(filename, lineages, raw, tracked):
-    """Saves raw, tracked, and lineage data into one trks_file.
+    """Saves raw, tracked, and lineage data from multiple movies into one trks_file.
 
     Args:
-        filename (str): full path to the final trk files.
-        lineages (dict): a list of dictionaries saved as a json.
+        filename (str or io.BytesIO): full path to the final trk files or bytes object
+            to save the data to
+        lineages (list): a list of dictionaries saved as a json.
         raw (np.array): raw images data.
         tracked (np.array): annotated image data.
 
     Raises:
         ValueError: filename does not end in ".trks".
     """
-    if not str(filename).lower().endswith('.trks'):
+    ext = os.path.splitext(str(filename))[-1]
+    if not isinstance(filename, io.BytesIO) and ext != '.trks':
         raise ValueError('filename must end with `.trks`. Found %s' % filename)
 
-    with tarfile.open(filename, 'w:gz') as trks:
+    save_track_data(filename=filename,
+                    lineages=lineages,
+                    raw=raw,
+                    tracked=tracked,
+                    lineage_name='lineages.json')
+
+
+def save_trk(filename, lineages, raw, tracked):
```

Comment:
For clarity id prefer argument to be `lineage` rather than `lineages`. It's a convention we've adopted as we've developed. If you do change it, don't forget to update in `tracking.py`

Reply:
Do you want to see the same change in `save_trks` and `save_track_data` or just in `save_trk`?

Reply:
`save_trks` should be lineages (plural) because there is more than one. `save_track_data` is more difficult. I think my preference is to leave it plural there as well to note that it is possible for there to be more than one.